### ISSUE TITLE
UnifiedNumberFormat notation: 'compact' format Error

### ISSUE THREAD

**Issue by MarvelSQ** [formatjs/formatjs#1670]
**Which package?**
@formatjs/intl-unified-numberformat

**Describe the bug**
UnifiedNumberFormat didn't adjust the compact

**To Reproduce**
```js
const UnitFormatter = new UnifiedNumberFormat('en', {
  maximumIntegerDigits: 3,
  minimumFractionDigits: 2,
  maximumFractionDigits: 2,
  useGrouping: false,
  style: 'decimal',
  notation: 'compact'
});

UnitFormatter.format(999.995) // should be 1.00K but got 1,000.00
UnitFormatter.format(999995000) // should be 1.00B but got 1,000.00M
```

**Comment by MarvelSQ** [comment]
these expected results are coming from **Intl.NumberFormat** with same config

**Comment by longlho** [comment]
related to #1634 I believe

**Comment by foubei** [comment]
Have you solved this issue?

**Comment by pyrocat101** [comment]
I think this is a spec bug. Will be fixed along with the next major version release very soon :)

**Comment by pyrocat101** [comment]
Should be fixed by #1684.
